include/ginkgo/core/stop/residual_norm_reduction.hpp

```diff
@@ -30,100 +30,5 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * the executor where the algorithm is executed.
- *
- * @note To use this stopping criterion there are some dependencies. The
- * constructor depends on `initial_residual` in order to compute the first
- * relative residual norm. The check method depends on either the
- * `residual_norm` or the `residual` being set. When any of those is not
- * correctly provided, an exception ::gko::NotSupported() is thrown.
- *
- * @ingroup stop
- */
-template <typename ValueType = default_precision>
-class ResidualNormReduction
-    : public EnablePolymorphicObject<ResidualNormReduction<ValueType>,
-                                     Criterion> {
-    friend class EnablePolymorphicObject<ResidualNormReduction<ValueType>,
-                                         Criterion>;
-
-public:
-    using NormVector = matrix::Dense<remove_complex<ValueType>>;
-    using Vector = matrix::Dense<ValueType>;
-
-    GKO_CREATE_FACTORY_PARAMETERS(parameters, Factory)
-    {
-        /**
-         * Relative residual norm goal
-         */
-        remove_complex<ValueType> GKO_FACTORY_PARAMETER(reduction_factor,
-                                                        1e-15);
-    };
-    GKO_ENABLE_CRITERION_FACTORY(ResidualNormReduction<ValueType>, parameters,
-                                 Factory);
-    GKO_ENABLE_BUILD_METHOD(Factory);
-
-protected:
-    bool check_impl(uint8 stoppingId, bool setFinalized,
-                    Array<stopping_status> *stop_status, bool *one_changed,
-                    const Criterion::Updater &) override;
-
-    explicit ResidualNormReduction(std::shared_ptr<const gko::Executor> exec)
-        : EnablePolymorphicObject<ResidualNormReduction, Criterion>(exec),
-          device_storage_{exec, 2}
-    {}
-
-    explicit ResidualNormReduction(const Factory *factory,
-                                   const CriterionArgs &args)
-        : EnablePolymorphicObject<ResidualNormReduction, Criterion>(
-              factory->get_executor()),
-          parameters_{factory->get_parameters()},
-          device_storage_{factory->get_executor(), 2}
-    {
-        if (args.initial_residual == nullptr) {
-            GKO_NOT_SUPPORTED(nullptr);
-        }
-
-        auto exec = factory->get_executor();
-
-        auto dense_r = as<Vector>(args.initial_residual);
-        starting_tau_ = NormVector::create(
-            exec, dim<2>{1, args.initial_residual->get_size()[1]});
-        u_dense_tau_ = NormVector::create_with_config_of(starting_tau_.get());
-        dense_r->compute_norm2(starting_tau_.get());
-    }
-
-private:
-    std::unique_ptr<NormVector> starting_tau_{};
-    std::unique_ptr<NormVector> u_dense_tau_{};
-    /* Contains device side: all_converged and one_changed booleans */
-    Array<bool> device_storage_;
-};
-
-
-}  // namespace stop
-}  // namespace gko
-
-
-#endif  // GKO_CORE_STOP_RESIDUAL_NORM_REDUCTION_HPP_
+// This file is deprecated and will be removed in a later release.
```

Comment:
Maybe try `#warning this file ....` as @tcojean said
I prefer use `in a later major release` or `in 2.0.0 release`